src/modules/hip/kernel/resample.hpp

```diff
@@ -0,0 +1,298 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+// -------------------- Set 0 - resample kernel device helpers  --------------------
+
+__device__ __forceinline__ float resample_hip_compute(float &x, float &scale, float &center, float *lookup, int &lookupSize)
+{
+    float locRaw = x * scale + center;
+    int locFloor = std::floor(locRaw);
+    float weight = locRaw - locFloor;
+    locFloor = std::max(std::min(locFloor, lookupSize - 2), 0);
+    float current = lookup[locFloor];
+    float next = lookup[locFloor + 1];
```

Comment:
please combine line 12 and 13 and compute diff
float diff = lookup[locFloor + 1] - lookup[locFloor];

Reply:
Hi Rajy,
The final return value from this function is 
```
current + weight * (next - current);
```

So if we store difference in separate variable as below
```
float diff = lookup[locFloor + 1] - lookup[locFloor];
```

Then we need to return final value as
return lookup[locFloor] + weight * diff

This requires the 3 memory access from lookup
to reduce this to 2 memory access, we stored them in separate variables in current code and return the final value as
```
    float current = lookup[locFloor];
    float next = lookup[locFloor + 1];
    return current + weight * (next - current);
```